lib/autorest.ts

```diff
@@ -0,0 +1,331 @@
+import * as commonmark from "commonmark";
+import * as jsYaml from "js-yaml";
+import * as os from "os";
+import { RunOptions, RunResult, runSync } from "./run";
+
+/**
+ * Options that can be passed to autorestExecutable().
+ */
+export interface AutoRestExecutableOptions {
+  /**
+   * The platform that AutoRest will be run on.
+   */
+  osPlatform?: string;
+  /**
+   * The path to the AutoRest executable. This can be either the folder that the executable is in or
+   * the path to the executable itself.
+   */
+  autorestPath?: string;
+}
+
+/**
+ * Get the executable that will be used to run AutoRest.
+ * @param options The options for specifying which executable to use.
+ */
+export function autorestExecutable(options: AutoRestExecutableOptions): string {
+  if (!options.osPlatform) {
+    options.osPlatform = os.platform();
+  }
+  let result: string = options.autorestPath || "";
+  if (!result.endsWith("autorest") && !result.endsWith("autorest.cmd")) {
+    if (result && !result.endsWith("/") && !result.endsWith("\\")) {
+      result += "/";
+    }
+    result += "autorest";
+  }
+  if (options.osPlatform === "win32" && !result.endsWith(".cmd")) {
+    result += ".cmd";
+  }
+  return result;
+}
+
+export type Switch = "" | boolean;
```

Comment:
Why?

Reply:
I kept typing `"" | boolean`, so I felt like it deserved a name. It does describe a command-line argument switch that can be described in a readme.md file.

Reply:
I get that. Why it can be an empty string? If it's defined like
`csharp` instead of `csharp: true` or something?

Reply:
Yup. In some swagger-to-sdk config files you'll see:
```json
{
  "python": ""
}
```
which is the equivalent of specifying `--python` on the command-line.